### PR TITLE
support std

### LINKED ISSUE #465: Handle std and prelude in name resolution
With the name resolution getting better, it's now more obvious that `std` is missing completely :) I guess we need to get the path to the rustup-installed std source code somehow. Of course that will also mean we have to parse a lot more code.

Once we have `std`, we should also import (basically) `std::prelude::v1::*` in every module.

### PR DESCRIPTION
**Pull request by matklad**
closes #465

### CODE REVIEW

File: crates/ra_lsp_server/tests/heavy_tests/main.rs [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
+    server.wait_for_feedback("workspace loaded");
+    eprintln!("loading took    {:?}", project_start.elapsed());
+    let completion_start = Instant::now();
+    let res = server.send_request::<Completion>(CompletionParams {
+        text_document: server.doc_id("src/lib.rs"),
+        context: None,
+        position: Position::new(0, 23),
+    });
+    assert!(format!("{}", res).contains("HashMap"));
+    eprintln!("completion took {:?}", completion_start.elapsed());
```
Comment (matklad) [inline crates/ra_lsp_server/tests/heavy_tests/main.rs]: For **debug** build I get

```
loading took    4.014003867s
completion took 6.368790446s
```

For release, 

```
loading took    303.160112ms
completion took 321.775276ms
```

Comment (DJMcNab) [reply crates/ra_lsp_server/tests/heavy_tests/main.rs]: That completion time seems unfortunately long.

Comment (matklad) [reply crates/ra_lsp_server/tests/heavy_tests/main.rs]: That's a one-time from scratch analysis. All subsequent completions are instant.

What we need to do ideally though is to kick basic analysis eagerly, after we've loaded the workspace, and not lazily on completion...

**Pull request merged**